content/en/integrations/faq/how-to-use-bean-regexes-to-filter-your-jmx-metrics-and-supply-additional-tags.md

```diff
@@ -0,0 +1,71 @@
+---
+title: How to use Bean Regexes to filter your JMX metrics and supply additional tags
+kind: faq
+further_reading:
+- link: "/integrations/java/"
+  tag: "Documentation"
+  text: "Java integration"
+- link: "/integrations/faq/view-jmx-data-in-jconsole-and-set-up-your-jmx-yaml-to-collect-them/"
+  tag: "Documentation"
+  text: "View JMX data in jConsole and set up your jmx.yaml to collect them"
+---
+
+Datadog supports regexes to match JMX Mbean names and domain names to configure your `include` and `exclude` filters. The regexes must conform to [Java’s regular expression format][2]. Note that these filters were added in version 5.5.0.
+
+Capture groups from the provided regex can be used to supply additional tag values for your metrics.
+
+This article provides one example of how to use the `bean_regex` from the [Java integration][3], and how to reference such capture groups to set additional tags.
+
+Suppose you have the following Mbean name: `domain.example.com:name=my.metric.name.env.dev.region.eu-central-1.method.GET.status.200`. There is some information we could use as tags once the Agent has collected the metric. For instance, we could export such metric with the following tags:
+
+* `env`: `dev`
+* `region`: `eu-central-1`
+* `method`: `GET`
+* `status`: `200`
+
+Bean regexes can be supplied as a single regex or a list of regex expressions. For the latter, only the first entry of the list that matches will be taken into account. See an example of a configuration file to export your custom metrics with some additional tags:
+
+```yaml
+init_config:
+  is_jmx: true
+
+instances:
+  - host: "<JMX_ENDPOINT>"
+    port: "<JMX_PORT>"
+
+    conf:
+      - include:
+          domain: domain.example.com
+          bean_regex:
+            - "domain.example.com:name=my.metric.name.*(?:\.env\.)([a-z]+)(?:.*\.region\.)([a-z-]+[0-9])(?:.*\.method\.)([A-Z]+)(?:.*\.status\.)([0-9]+)(?:.*)"
```

Comment:
Thanks for your feedback @olivielpeau!

Regarding the backslashes, are you concerned about the YAML visualization in DD pages or the DD agent itself handling these slashes? If this is an agent matter, I'd say that it might work fine. I've tested this out and it worked as expected. Either way, I'll be happy to modify it as you suggested. Let me know your thoughts.

For checking if the format is ok for publishing, if someone from de DD organization pulls this PR and pushes it to some branch in this repo, we'll be able to visualize it using https://docs-staging.datadoghq.com/<branch-name> I suppose. Am I right?

And hello @kayayarai! I totally agree about link this page in the further reading section of the main JMXFetch docs page. If you think that this is ok, I'd really appreciate that 😃

Reply:
@olivielpeau, the merge happened too fast, so I didn't have the chance to hear from you. I'm sorry for that.

Could you please check if the #10553 fixes our YAML loading error?